(In Code Review) Add all opportunity based endpoints

mattkranzler5 opened the pull request:
https://jira.desk.technology/browse/APIJ-39

mattkranzler5 committed:
• Added company count to filter
• Added fields to getCompanyFilter endpoints

mattkranzler5 committed:
• Added OpportunityStage & OpportunityStageType model objects
• Added OpportunityStageService & RxOpportunityStageService
• Fixed javadoc error in OutboundMailboxService

mattkranzler5 committed:
• Added opportunity filters endpoint to both filter services

mattkranzler5 committed:
Merge branch 'develop' into tickets/APIJ-39-add-all-opportunity-based

mattkranzler5 committed:
• Added company label type
• Added opportunity model object
• Added opportunity services

mattkranzler5 committed:
Added get opportunty endpoints

mattkranzler5 committed:
• Added system event models
• Added opportunity activity models
• Added endpoints to get opportunity timeline and activities
• Added custom type adapter to parse objects from the opportunity activities endpoint since it returns a list of objects with different types
• Added unit tests for custom type adapter

mattkranzler5 committed:
Merge branch 'tickets/APIJ-43-add-fields-including' into tickets/APIJ-39-add-all-opportunity-based

jerrellmardis commented on api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java:
```diff
@@ -0,0 +1,131 @@
+    private int probability;
+    private Date createdAt;
+    private Date updatedAt;
+    private BigDecimal amount;
```
Why `BigDecimal`?

mattkranzler5 replied:
For currency amounts it is recommended to use `BigDecimal` in java. See http://stackoverflow.com/questions/3730019/why-not-use-double-or-float-to-represent-currency

jerrellmardis replied:
:+1:

jerrellmardis commented on api-client/src/main/java/com/desk/java/apiclient/model/Opportunity.java:
```diff
@@ -0,0 +1,131 @@
+    private String[] labels;
+    private int[] labelIds;
+    private Date closeDate;
+    private HashMap<String, String> customFields;
```
Should use `Map` instead.

jerrellmardis commented on api-client/src/main/java/com/desk/java/apiclient/model/OpportunityStage.java:
```diff
@@ -0,0 +1,78 @@
+    private String name;
+    private int probability;
+    private OpportunityStageType opportunityStageType;
+    private Links _links;
```
`_links` should be `links` with the SerializedName of `_links`

jerrellmardis commented:
First pass done

mattkranzler5 committed:
• Changed custom fields to be of type `Map` in `Opportunity`
• Changed `_links` to be `links` and have a `@SerializedName(_links)` annotation in various objects

mattkranzler5 committed:
Fixed some okhttp 3 references
